packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -2669,7 +2669,9 @@ class DiagnosticsProperty<T> extends DiagnosticsNode {
     json['defaultLevel'] = describeEnum(_defaultLevel);
     if (value is Diagnosticable || value is DiagnosticsNode)
       json['isDiagnosticableValue'] = true;
-    if (value is num || value is String || value is bool || value == null)
+    if (v is num)
+      json['value'] = v.isFinite ? v :  v.toString();
```

Comment:
Add a comment explaining this is due to json restrictions.

Reply:
done